### DISCUSSION TITLE
Correct syntax of dependencies section?

### DISCUSSION THREAD

**Discussion by JoergBruenner** [DependencyTrack/dependency-track#2447] [Q&A] [answered]
Hi all,

I' m building a SBom by hand. the import works, the dependency graph isn't shown the expected way:

My application depends on web-framework and persistence. The dependency graph graph shows only the first lib in the
dependencies section and its dependencies. If I change the order of web-framework and persistence , the graph shows the 
changed first library and their dependencies. What's my fault?

Thanks a lot for your help
Joerg

The bom is:
{
    "bomFormat": "CycloneDX",
    "specVersion": "1.4",
    "serialNumber": "urn:uuid:e1322ee8-d940-4ebc-a26e-25e1c0b6e705",
    "version": 1,
    "metadata": {
        "component": {
            "bom-ref": "pkg:maven/org.acme/BlaBLubb@1.0",
            "type": "application",
            "group": "org.acme",
            "name": "BlaBlubb",
            "version": "1.0",
            "purl": "pkg:maven/org.acme/BlaBLubb@1.0"
        }
    },
    "components": [
        {
            "bom-ref": "pkg:maven/org.acme/common-util@3.0.0",
            "type": "library",
            "group": "org.acme",
            "name": "common-util",
            "version": "3.0.0",
            "purl": "pkg:maven/org.acme/common-util@3.0.0"
        },
        {
            "bom-ref": "pkg:maven/org.acme/common-util@3.0.1",
            "type": "library",
            "group": "org.acme",
            "name": "common-util",
            "version": "3.0.1",
            "purl": "pkg:maven/org.acme/common-util@3.0.1"
        },
        {
            "bom-ref": "pkg:maven/org.acme/persistence@3.1.0",
            "type": "library",
            "group": "org.acme",
            "name": "persistence",
            "version": "3.1.0",
            "purl": "pkg:maven/org.acme/persistence@3.1.0"
        },
        {
            "bom-ref": "pkg:maven/org.acme/web-framework@1.0.0",
            "type": "library",
            "group": "org.acme",
            "name": "web-framework",
            "version": "1.0.0",
            "purl": "pkg:maven/org.acme/web-framework@1.0.0"
        }
    ],
    "dependencies": [
        {
            "ref": "pkg:maven/org.acme/BlaBLubb@1.0",
            "dependsOn": [
                "pkg:maven/org.acme/web-framework@1.0.0"
            ]
        },
        {
            "ref": "pkg:maven/org.acme/BlaBLubb@1.0",
            "dependsOn": [
                "pkg:maven/org.acme/persistence@3.1.0"
            ]
        },
        {
            "ref": "pkg:maven/org.acme/persistence@3.1.0",
            "dependsOn": [
                "pkg:maven/org.acme/common-util@3.0.0"
            ]
        },
        {
            "ref": "pkg:maven/org.acme/common-util@3.0.0"
        },
        {
            "ref": "pkg:maven/org.acme/web-framework@1.0.0",
            "dependsOn": [
                "pkg:maven/org.acme/common-util@3.0.1"
            ]
        },
        {
            "ref": "pkg:maven/org.acme/common-util@3.0.1"
        }
    ]
}

**Comment by nscuro** [comment, marked as answer]
Your graph contains a node for the main component `pkg:maven/org.acme/BlaBLubb@1.0` twice:

```json
{
    "ref": "pkg:maven/org.acme/BlaBLubb@1.0",
    "dependsOn": [
        "pkg:maven/org.acme/web-framework@1.0.0"
    ]
},
{
    "ref": "pkg:maven/org.acme/BlaBLubb@1.0",
    "dependsOn": [
        "pkg:maven/org.acme/persistence@3.1.0"
    ]
},
```

`dependsOn` is an array, so you should specify *all* direct dependencies of `pkg:maven/org.acme/BlaBLubb@1.0` in a single node:

```json
{
    "ref": "pkg:maven/org.acme/BlaBLubb@1.0",
    "dependsOn": [
        "pkg:maven/org.acme/persistence@3.1.0",
        "pkg:maven/org.acme/web-framework@1.0.0"
    ]
},
```

**Reply by JoergBruenner** [reply]
Great hint, now it works :-) Thanks a lot Joerg.
